site/components/pages/Retirements/List/Item.tsx

```diff
@@ -0,0 +1,51 @@
+import { FC } from "react";
+import { useRouter } from "next/router";
+import Link from "next/link";
+
+import { Text } from "@klimadao/lib/components";
+
+import { KlimaRetire } from "@klimadao/lib/types/subgraph";
+import { concatAddress, getTypeofTokenByAddress } from "@klimadao/lib/utils";
+import ArrowForwardIcon from "@mui/icons-material/ArrowForwardOutlined";
+import { LeafIcon } from "./Leaf";
+
+import * as styles from "./styles";
+
+type Props = {
+  retirement: KlimaRetire;
+};
+
+export const RetirementItem: FC<Props> = (props) => {
+  const { retirement } = props;
+  const { locale } = useRouter();
+
+  const retirementDate = new Date(parseInt(retirement.timestamp) * 1000); //expects milliseconds
+  const formattedDate = new Intl.DateTimeFormat(locale, {
+    dateStyle: "full",
+  }).format(retirementDate);
+  const typeOfToken = getTypeofTokenByAddress(retirement.offset.tokenAddress);
+  // typeOfToken can be undefined! Due to incorrect tokenAddress received from subgraph - Why?
+
+  const url = `/retirements/${retirement.beneficiaryAddress}/${
+    Number(retirement.index) + 1
+  }`;
+
+  return (
+    <Link href={url} passHref>
+      <a className={styles.allRetirementsListItem}>
+        <LeafIcon />
```

Comment:
how about we show the icon for the token that was retired?

Reply:
Good idea. But the token address has to match. Otherwise this doesn't work.
Lets talk about the graph data again

Reply:
Found it! The subgraph calls it "pool"
Offset "tokenAddress" seems to be something else then.

Done